src/main/java/org/mockito/internal/matchers/CompareTo.java

```diff
@@ -9,24 +9,27 @@
 
 import java.io.Serializable;
 
-
 public abstract class CompareTo<T extends Comparable<T>> implements ArgumentMatcher<T>, Serializable {
-    private final Comparable<T> wanted;
+    private final T wanted;
 
-    public CompareTo(Comparable<T> value) {
+    public CompareTo(T value) {
         this.wanted = value;
     }
 
-    @SuppressWarnings("unchecked")
     public boolean matches(T actual) {
-        return matchResult(((Comparable)actual).compareTo(wanted));
+        if (actual == null) {
+            return false;
+        }
+       
```

Comment:
To avoid a `ClassCastException` it must be ensured that _wanted_ is compatible to _actual_ : 
```
if (!actual.getClass().isInstance(wanted)){ 
   return false
}
```